go/vt/vtctl/reparentutil/planned_reparenter.go

```diff
@@ -518,6 +518,11 @@ func (pr *PlannedReparenter) reparentShardLocked(
 		return err
 	}
 
+	err = pr.verifyAllTabletsReachable(ctx, tabletMap)
+	if err != nil {
+		return err
+	}
```

Comment:
We can run this even before we acquire the shard lock... I don't have strong opinions.

Reply:
Even though we don't need the lock, if we hold it we prevent something else from changing things out from under us while we verify the reachability, so I'm in favor of keeping it here